File: 2206. Divide Array Into Equal Pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        mp = {}

        for i in nums:
            mp[i] = mp.get(i, 0) + 1
        for key, val in mp.items():
            if val % 2 != 0:
                return False
        return True

File: 2206. Divide Array Into Equal Pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();

        for (int i : nums) {
            mp.put(i, mp.getOrDefault(i, 0) + 1);
        }
        for (int val : mp.values()) {
            if (val % 2 != 0) {
                return false;
            }
        }
        return true;
    }
}
